src/metabase/models/setting.clj

```diff
@@ -384,14 +384,22 @@
   "Get the value, if any, of `setting-definition-or-name` from the DB (using / restoring the cache as needed)."
   ^String [setting-definition-or-name]
   (let [setting (resolve-setting setting-definition-or-name)]
-    (when (allows-site-wide-values? setting)
-      (let [v (if *disable-cache*
-                (db/select-one-field :value Setting :key (setting-name setting-definition-or-name))
-                (do
-                  (setting.cache/restore-cache-if-needed!)
-                  (clojure.core/get (setting.cache/cache) (setting-name setting-definition-or-name))))]
-        (when (seq v)
-          v)))))
+    ;; cannot use db (and cache populated from db) if db is not set up
+    (when-let [db-is-set-up? (resolve 'metabase.db/db-is-set-up?)]
+      (when (db-is-set-up?)
+        (when (allows-site-wide-values? setting)
+          (let [v (if *disable-cache*
+                    (db/select-one-field :value Setting :key (setting-name setting-definition-or-name))
+                    (do
+                      (setting.cache/restore-cache-if-needed!)
+                      (let [cache (setting.cache/cache)]
```

Comment:
Would it be better just to have a `ReentrantReadWriteLock` around the cache or something like that? So we can have as many readers as we want simultaneously but when the cache is getting repopulated it will block anyone from reading until the repopulation finishes. Maybe it doesn't matter if this is only during initial startup

Reply:
Yeah it is just during the initial population. Let's leave it like this for this change and we can improve it in the future if we like. Don't want to introduce another bug while fixing.